Source/JavaScriptCore/runtime/DateConstructor.cpp

```diff
@@ -71,6 +71,37 @@ static double millisecondsFromComponents(JSGlobalObject* globalObject, const Arg
     VM& vm = globalObject->vm();
     auto scope = DECLARE_THROW_SCOPE(vm);
 
+    auto toIntegerOrInfinity = [](double d) {
+        return trunc(std::isnan(d) ? 0.0 : d + 0.0);
+    };
```

Comment:
If we already confirmed `isfinite`, can't this be inlined with the assumption that it's not `NaN`?

Reply:
I wonder if we should keep it since `std::isnan` is super cheap anyway, and it makes `toIntegerOrInfinity` aligned to its implementation.